code/scripting/api/libs/base.cpp

```diff
@@ -62,12 +62,23 @@ ADE_FUNC(rand32,
 	int numargs = ade_get_args(L, "|ii", &a, &b);
 
 	int result;
-	if (numargs == 2)
-		result = Random::next(a, b);
-	else if (numargs == 1)
-		result = Random::next(a);
-	else
+	if (numargs == 2) {
+		if (a <= b) {
+			result = Random::next(a, b);
+		} else {
+			Warning(LOCATION, "rand32() script function was passed an invalid range (%d ... %d)!", a, b);
```

Comment:
This and the other should be a `LuaError` since that will give the script user a stack trace in their script.